### DISCUSSION TITLE
How can i disable blink completions in an inline custom prompt?

### DISCUSSION THREAD

**Discussion by xheisenbugx** [olimorris/codecompanion.nvim#547] [Q&A] [answered]
Hi everyone,

I'm using LazyVim along with the Blink and Snacks plugins. I noticed that when an input is required for a custom prompt, the Blink plugin activates, and it becomes super annoying as it tries to autocomplete automatically.

I assume this might also happen with nvim-cmp, but I still need to confirm that.

Has anyone encountered this issue, or does anyone know how to fix it?

Thanks in advance!

**Comment by olimorris** [comment]
Yep this is happening for me as well. There may be a setting to disable it for certain file types. I haven't looked into it yet.

**Reply by olimorris** [reply]
Straight from the blink README:

```lua

 -- Example for blocking multiple filetypes
enabled = function()
  return not vim.tbl_contains({ "lua", "markdown" }, vim.bo.filetype) and vim.bo.buftype ~= "prompt"
end,
```

We can put this in our config for the plugin.

**Comment by bassamsdata** [comment]
A different approach from the one Oli suggested is this: if you're using [snacks.input](https://github.com/folke/snacks.nvim/blob/main/docs/input.md), you can do the following:  
```lua
input = { 
  b = {
    completion = false, -- disable blin completions in input
  }
}
```

**Reply by xheisenbugx** [reply]
I gave it a try, but it didn’t work. I suspect the issue is that CodeCompanion is relying on vim.ui.input instead of Snacks.input, i will try the suggested approach of Oli

**Reply by bassamsdata** [reply, marked as answer]
Actually, `snacks.input` is just a wrapper around `vim.ui.input()`. So, if you're typing in the prompt inside a floating window like the one below, you're using `snacks.input`.  
<img width="765" alt="Screenshot 2024-12-17 at 8 11 30 PM" src="https://github.com/user-attachments/assets/730c2bb3-e7a7-447e-9f73-b5bfbf0c3328" />


One suggestion to Oli's solution is to add the filetype `"snacks_input"` to the filetypes list, as `snacks.input` uses this filetype by default.

**Reply by xheisenbugx** [reply]
It works! 🎉

Here's the configuration I used, in case it helps someone else, it also removes auto brackets in codecompanion provider, otherwise it will add some extra parentheses, which is also annoying

```
{
    "saghen/blink.cmp",
    dependencies = { "olimorris/codecompanion.nvim", "saghen/blink.compat" },
    opts = {
      enabled = function()
        return vim.bo.buftype ~= "prompt" and vim.b.completion ~= false
      end,
      completion = {
        accept = {
          auto_brackets = {
            kind_resolution = {
              blocked_filetypes = { "typescriptreact", "javascriptreact", "vue", "codecompanion" },
            },
          },
        },
      },
      sources = {
        compat = { "codecompanion" },
        providers = {
          codecompanion = {
            name = "CodeCompanion",
            module = "codecompanion.providers.completion.blink",
            enabled = true,
          },
        },
      },
    },
  }
```

**Comment by olimorris** [comment]
I should have said you'd need to customise this for your own config. Thanks @bassamsdata
